### PR TITLE
Script to enforce copyright header

### PR DESCRIPTION
**Pull request by ricealexanderb**
Added a script that makes sure there's a standard copyright header on all swift files in the repo, and then made the release-candidate workflow use it the same way it uses the changelog entry script.

If this seems like a good way to do header enforcement, I can apply this change across all the other repos.

### CODE REVIEW

**Review by CB-HaoxinLi — DISMISSED** [review]
LGTM

File: Scripts/copyright.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+file_extensions_accepted = ['.h', '.m', '.swift']
+
+# Note: This list is not meant to be exhaustive. 
+exclusions = %w[Scripts .github]
+
+# Find all resources in the Chartboost Mediation repo that match the accepted file extensions
+files = []
+
+# We expect to be in the Scripts folder so go one level up to the root of the repo
+Find.find("..") do |path|
```
[the remark this answers is not in the document]
Comment (ricealexanderb) [reply Scripts/copyright.rb]: Fixed

**Comment by ricealexanderb** [comment]
> I think it's replacing all the occurrences of comment blocks instead of the ones at the beginning of the file.

It was using `gsub()` (replace globally) where it should have just used `sub()` (replace first)

I also found a bug where the file handle wasn't necessarily at the expected spot in the file, so instead of using the same handle for read/write I used a temp file, which I also like better because it makes the changes atomic.

**Review by daniel-barros — DISMISSED** [review]
Tested it in another branch and it worked great!
Only one minor detail, it could be nice to add an empty line at the end of the headers.
Otherwise, if a file starts like
```
import Foundation
...
```

the script adds:
```
// Copyright 2022-2023 Chartboost, Inc.
//
// Use of this source code is governed by an MIT-style
// license that can be found in the LICENSE file.
import Foundation
...
```

File: Scripts/copyright.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      ensure
+        tmp.close
+        tmp.unlink
+      end
+    end
+  end
+end
+
+if files_have_been_changed
+    puts "changes"
```
Comment (kendallrogers) [inline Scripts/copyright.rb:64]: I wonder if we'd be better outputting the paths of files changed, then adding those with `git add` in the Push Header Changes step. This way, if there were some other file changes for some odd reason, they wouldn't get committed. Not a huge deal though.

**Review by daniel-barros — APPROVED** [review]

**Pull request closed without merging**